x-pack/plugins/infra/public/alerting/log_threshold/components/expression_editor/hooks/use_chart_preview_data.tsx

```diff
@@ -43,6 +46,55 @@ export const useChartPreviewData = ({
       cancelPreviousOn: 'creation',
       createPromise: async () => {
         setHasError(false);
+        if (isRatioRule(ruleParams.criteria)) {
+          const ratio = await Promise.all([
+            callGetChartPreviewDataAPI(
+              logViewReference,
+              http!.fetch,
+              { ...ruleParams, criteria: [...ruleParams.criteria[0]] },
+              buckets,
+              executionTimeRange
+            ),
+            callGetChartPreviewDataAPI(
+              logViewReference,
+              http!.fetch,
+              { ...ruleParams, criteria: [...ruleParams.criteria[1]] },
+              buckets,
+              executionTimeRange
+            ),
+          ]);
+          // The two array have the same length and the same time range.
+          let seriesQueryA = ratio[0].data.series[0].points;
+          let seriesQueryB = ratio[1].data.series[0].points;
+
+          // When groupBy and a filter is applied, return the ratio only for the filtered grouped-by
+          if (ruleParams.groupBy.length && filterSeriesByGroupName) {
+            seriesQueryA =
+              ratio[0].data.series.find((series) => series.id === filterSeriesByGroupName)
+                ?.points || [];
+
+            seriesQueryB =
+              ratio[1].data.series.find((series) => series.id === filterSeriesByGroupName)
+                ?.points || [];
+          }
```

Comment:
Based on this snippet, if the Id has no match with the passed `filterSeriesByGroupName`, it will fallback to an empty array, but before they are set to the first series points. What is the expected behaviour, an empty array or keeping the first series points?

Reply:
@tonyghiani, If I get your question, yes, in the default value, we consider that the rule has no `groupBy`, so we use the first (only) series.
If we detect that `groupBy` is passed, we start looking into the series to match the groupBy key.